cluster-autoscaler/core/scale_down.go

```diff
@@ -628,7 +618,7 @@ func (sd *ScaleDown) isNodeBelowUtilzationThreshold(node *apiv1.Node, utilInfo s
 // updateUnremovableNodes updates unremovableNodes map according to current
 // state of the cluster. Removes from the map nodes that are no longer in the
 // nodes list.
-func (sd *ScaleDown) updateUnremovableNodes(nodes []*apiv1.Node) {
+func (sd *ScaleDown) updateUnremovableNodes(nodes []*schedulernodeinfo.NodeInfo) {
```

Comment:
nit: It's probably not super important I think passing Node was more elegant than NodeInfo. The way I see it NodeInfo is a Node + list of pods, if we don't care about pods we should use Node directly.

Reply:
Returning list of Nodes requires snapshot to either build it on demand each time, or maintain two cached lists. It seemed simpler to just reuse NodeInfos here.

Reply:
See above, got rid of GetAllNodes method